datafusion-cli/src/catalog.rs

```diff
@@ -145,16 +146,24 @@ impl SchemaProvider for DynamicFileSchemaProvider {
         self.inner.register_table(name, table)
     }
 
-    async fn table(&self, name: &str) -> Option<Arc<dyn TableProvider>> {
-        let inner_table = self.inner.table(name).await;
+    async fn table(
+        &self,
+        name: &str,
+    ) -> Result<Option<Arc<dyn TableProvider>>, DataFusionError> {
```

Comment:
The rest of the file uses this style, so I think the `DataFusionError` is unecessary

```suggestion
    ) -> Result<Option<Arc<dyn TableProvider>>> {
```

Reply:
done